Failed to import native files

Hello Wollok Team,

When I try to add a new native wollok file in `org.uqbar.project.wollok.lib/src/wollok.wollokmf`:
In this case `game.wlk` (in the same location as wollok.wlk)

![screenshot from 2015-09-26 20-44-50](https://cloud.githubusercontent.com/assets/1645850/10120565/dd4f56f6-648f-11e5-8bcc-84ad9cc98f30.png)

Then I run the environment, and the objects than I have added in `org.uqbar.project.wollok.lib/src/game.wlk` don't exist in the global context.

The eclipse application console prints the following stack trace:

```
org.eclipse.xtext.resource.ClasspathUriResolutionException: classpath:/game.wlk
	at org.eclipse.xtext.resource.XtextResourceSet$1.normalize(XtextResourceSet.java:250)
	at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:203)
	at org.eclipse.xtext.resource.SynchronizedXtextResourceSet.getResource(SynchronizedXtextResourceSet.java:24)
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider.loadResource(WollokGlobalScopeProvider.java:125)
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$5.apply(WollokGlobalScopeProvider.java:114)
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$5.apply(WollokGlobalScopeProvider.java:1)
	at org.eclipse.xtext.xbase.lib.internal.FunctionDelegate.apply(FunctionDelegate.java:41)
	at com.google.common.collect.Lists$TransformingRandomAccessList.get(Lists.java:572)
	at java.util.AbstractList$Itr.next(AbstractList.java:358)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:645)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:645)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at org.eclipse.xtext.scoping.impl.ImportScope.getLocalElementsByName(ImportScope.java:154)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getElements(AbstractScope.java:128)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getElements(AbstractScope.java:131)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getElements(AbstractScope.java:131)
	at org.eclipse.xtext.scoping.impl.ScopeBasedSelectable.getExportedObjects(ScopeBasedSelectable.java:37)
	at org.eclipse.xtext.scoping.impl.ImportScope.getLocalElementsByName(ImportScope.java:152)
	at org.eclipse.xtext.scoping.impl.ImportScope.getSingleLocalElementByName(ImportScope.java:140)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:102)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getLinkedObjects(DefaultLinkingService.java:120)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:248)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:224)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getEObject(ResourceSetImpl.java:223)
	at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:199)
	at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:259)
	at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eResolveProxy(BasicEObjectImpl.java:1473)
	at org.uqbar.project.wollok.wollokDsl.impl.WVariableReferenceImpl.getRef(WVariableReferenceImpl.java:72)
	at org.uqbar.project.wollok.ui.highlight.WollokHighlightingCalculator._highlight(WollokHighlightingCalculator.java:46)
	at org.uqbar.project.wollok.ui.highlight.WollokHighlightingCalculator.highlight(WollokHighlightingCalculator.java:105)
	at org.uqbar.project.wollok.ui.highlight.WollokHighlightingCalculator.highlightElement(WollokHighlightingCalculator.java:28)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.DefaultSemanticHighlightingCalculator.highlightElementRecursively(DefaultSemanticHighlightingCalculator.java:81)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.DefaultSemanticHighlightingCalculator.searchAndHighlightElements(DefaultSemanticHighlightingCalculator.java:74)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.DefaultSemanticHighlightingCalculator.doProvideHighlightingFor(DefaultSemanticHighlightingCalculator.java:65)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.DefaultSemanticHighlightingCalculator.provideHighlightingFor(DefaultSemanticHighlightingCalculator.java:42)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.MergingHighlightedPositionAcceptor.provideHighlightingFor(MergingHighlightedPositionAcceptor.java:51)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler.reconcilePositions(HighlightingReconciler.java:139)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler.modelChanged(HighlightingReconciler.java:357)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler.modelChanged(HighlightingReconciler.java:331)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler$2$1.process(HighlightingReconciler.java:317)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler$2$1.process(HighlightingReconciler.java:1)
	at org.eclipse.xtext.util.concurrent.IUnitOfWork$Void.exec(IUnitOfWork.java:36)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:494)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.readOnly(XtextDocument.java:465)
	at org.eclipse.xtext.ui.editor.model.XtextDocument.readOnly(XtextDocument.java:105)
	at org.eclipse.xtext.ui.editor.syntaxcoloring.HighlightingReconciler$2.run(HighlightingReconciler.java:314)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
```


Strangely for `wollok.wlk` the file is in another location: `platform:/resource/.org.eclipse.jdt.core.external.folders/.link4/wollok.wlk`.

But WollokManifest.xtend produces two simiar URIs:
`"classpath:/wollok.wlk"`
`"classpath:/game.wlk"`

I tried creating a new file "json.wlk"

```

object json {

	method parse(aString) {
		return "lalala"
	}
	
	method format(anObject) {
		return "a json"
	}

}
```

Then addedd that to "wollok.wollokmf"
```
wollok.wlk
json.wlk
```

Then started wollok IDE, and created a new program like this

```
import json.json

program abc {

	console.println(json.parse('Hello Wollok'))

}
```

Without the import it was not able to resolve the "json" variable.
Then imports the linker worked fine.
But when I run the program I get this runtime error

```
ERROR [main] (LazyLinkingResource.java:228) - resolution of uriFragment '|0' failed.
org.uqbar.project.wollok.interpreter.WollokRuntimeException: Could NOT find resource 'json.json'
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$3.apply(WollokGlobalScopeProvider.java:80)
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$3.apply(WollokGlobalScopeProvider.java:1)
	at org.eclipse.xtext.xbase.lib.internal.FunctionDelegate.apply(FunctionDelegate.java:41)
	at com.google.common.collect.Lists$TransformingRandomAccessList.get(Lists.java:572)
	at com.google.common.collect.Lists$TransformingRandomAccessList.get(Lists.java:572)
	at java.util.AbstractList$Itr.next(AbstractList.java:358)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:645)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at org.eclipse.xtext.scoping.impl.ImportScope.getLocalElementsByName(ImportScope.java:154)
	at org.eclipse.xtext.scoping.impl.ImportScope.getSingleLocalElementByName(ImportScope.java:140)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:102)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getLinkedObjects(DefaultLinkingService.java:120)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:248)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:224)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.doResolveLazyCrossReference(LazyLinkingResource.java:191)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReference(LazyLinkingResource.java:150)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReferences(LazyLinkingResource.java:136)
	at org.eclipse.xtext.EcoreUtil2.resolveLazyCrossReferences(EcoreUtil2.java:513)
	at org.eclipse.xtext.validation.ResourceValidatorImpl.resolveProxies(ResourceValidatorImpl.java:159)
	at org.eclipse.xtext.validation.ResourceValidatorImpl.validate(ResourceValidatorImpl.java:68)
	at org.uqbar.project.wollok.launch.WollokChecker.validate(WollokChecker.java:220)
	at org.uqbar.project.wollok.launch.WollokChecker.validate(WollokChecker.java:183)
	at org.uqbar.project.wollok.launch.WollokChecker.parse(WollokChecker.java:110)
	at org.uqbar.project.wollok.launch.WollokChecker.doMain(WollokChecker.java:80)
	at org.uqbar.project.wollok.launch.WollokLauncher.main(WollokLauncher.java:29)
org.eclipse.emf.common.util.WrappedException: org.uqbar.project.wollok.interpreter.WollokRuntimeException: Could NOT find resource 'json.json'ERROR [main] (WollokChecker.java:87) - org.uqbar.project.wollok.interpreter.WollokRuntimeException: Could NOT find resource 'json.json'

	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:231)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.doResolveLazyCrossReference(LazyLinkingResource.java:191)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReference(LazyLinkingResource.java:150)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReferences(LazyLinkingResource.java:136)
	at org.eclipse.xtext.EcoreUtil2.resolveLazyCrossReferences(EcoreUtil2.java:513)
	at org.eclipse.xtext.validation.ResourceValidatorImpl.resolveProxies(ResourceValidatorImpl.java:159)
	at org.eclipse.xtext.validation.ResourceValidatorImpl.validate(ResourceValidatorImpl.java:68)
	at org.uqbar.project.wollok.launch.WollokChecker.validate(WollokChecker.java:220)
	at org.uqbar.project.wollok.launch.WollokChecker.validate(WollokChecker.java:183)
	at org.uqbar.project.wollok.launch.WollokChecker.parse(WollokChecker.java:110)
	at org.uqbar.project.wollok.launch.WollokChecker.doMain(WollokChecker.java:80)
	at org.uqbar.project.wollok.launch.WollokLauncher.main(WollokLauncher.java:29)
Caused by: org.uqbar.project.wollok.interpreter.WollokRuntimeException: Could NOT find resource 'json.json'
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$3.apply(WollokGlobalScopeProvider.java:80)
	at org.uqbar.project.wollok.scoping.WollokGlobalScopeProvider$3.apply(WollokGlobalScopeProvider.java:1)
	at org.eclipse.xtext.xbase.lib.internal.FunctionDelegate.apply(FunctionDelegate.java:41)
	at com.google.common.collect.Lists$TransformingRandomAccessList.get(Lists.java:572)
	at com.google.common.collect.Lists$TransformingRandomAccessList.get(Lists.java:572)
	at java.util.AbstractList$Itr.next(AbstractList.java:358)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:645)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at org.eclipse.xtext.scoping.impl.ImportScope.getLocalElementsByName(ImportScope.java:154)
	at org.eclipse.xtext.scoping.impl.ImportScope.getSingleLocalElementByName(ImportScope.java:140)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:102)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.scoping.impl.AbstractScope.getSingleElement(AbstractScope.java:105)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getLinkedObjects(DefaultLinkingService.java:120)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:248)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:224)
	... 11 more

```

Which is not actually the same as Charly reported, but maybe related

--- a/org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/WollokGlobalScopeProvider.xtend
+++ b/org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/WollokGlobalScopeProvider.xtend
@@ -34,17 +34,23 @@ class WollokGlobalScopeProvider extends DefaultGlobalScopeProvider {
 		val resourceSet = context.resourceSet
 
 		val exportedObjects = manifestFinder.allManifests(resourceSet).map[handleManifest(resourceSet)].flatten
-		val explicitImportedObjects = this.importedObjects(context)
+		val explicitImportedObjects = this.importedObjects(context, exportedObjects)
 
 		val defaultScope = super.getScope(parent, context, ignoreCase, type, filter)
 		new SimpleScope(new SimpleScope(defaultScope, exportedObjects), explicitImportedObjects)
 	}
 	
-	def importedObjects(Resource resource) {
+	def importedObjects(Resource resource, Iterable<IEObjectDescription> objectsFromManifests) {
 		val rootObject = resource.contents.get(0)
 		val imports = rootObject.getAllContentsOfType(Import)
 		val importedNamespaces = imports.map[importedNamespace]
-		importedNamespaces.map[name|
+		importedNamespaces
+		.filter[name|
+			// filters those imported from libraries, we won't handle them as regular imports
+			// this needs further works if we add more complex imports like *, or aliases.
+			!objectsFromManifests.exists[o| o.qualifiedName.toString == name]
+		]
+		.map[name|
 			var uri = generateUri(resource, name)
 			val r = EcoreUtil2.getResource(resource, uri)
 			if (r == null) throw new WollokRuntimeException("Could NOT find resource '" + name +"'");

